src/api/meetup/queries.server.ts

```diff
@@ -0,0 +1,177 @@
+ *
+ * @see {@link MeetupQueryConfig}
+ */
+export async function fetchMeetupEventById(
+  eventId: string,
+  config?: MeetupQueryConfig<MeetupEventType | null>
+) {
+  return queryCacheClient.fetchQuery({
+    ...config,
+    queryKey: ['fetchMeetupEventById', eventId],
+    async queryFn({ queryKey }) {
+      const data = await meetupGraqhqlClient.request<{
+        event: MeetupEventType | null;
+      }>({
+        document: EVENT_BY_EVENTID,
+        variables: { eventId: queryKey[1] },
+      });
+
+      return data.event;
+    },
+  });
+}
+
+/**
+ * ### Usage
+ *
+ * ```ts
+ * import { fetchRomajsMeetupsCount } from '@api/meetup/queries.server';
+ *
+ * const romaJsMeetupsCount: number = await fetchRomajsMeetupsCount();
+ * ```
+ *
+ * @see {@link MeetupQueryConfig}
+ */
+export async function fetchRomajsMeetupsCount(
+  config?: MeetupQueryConfig<number>
+): Promise<number> {
+  return queryCacheClient.fetchQuery({
+    ...config,
+    queryKey: ['fetchRomajsMeetupsCount'],
+    async queryFn() {
+      const data = await meetupGraqhqlClient.request<{
+        groupByUrlname: { pastEvents: { count: number } };
+      }>({
+        document: MEETUP_EVENTS_COUNT,
+        variables: {
+          urlname: import.meta.env.MEETUP_GROUP_ID,
+        },
+      });
+
+      const {
+        groupByUrlname: {
+          pastEvents: { count },
+        },
+      } = data;
+      return count;
+    },
+  });
+}
+
+/**
+ * ### Usage
+ *
+ * ```ts
+ * import { fetchNextUpcomingRomajsEvent } from '@api/meetup/queries.server';
+ *
+ * const upcomingRomaJsEvents = await fetchNextUpcomingRomajsEvent();
+ * ```
+ */
+export async function fetchUpcomingRomajsEvents(
+  config?: MeetupQueryConfig<MeetupEventType[] | null>
+): Promise<MeetupEventType[] | null> {
+  return queryCacheClient.fetchQuery({
+    ...config,
+    queryKey: ['fetchUpcomingRomajsEvents'],
+    async queryFn() {
+      const data = await meetupGraqhqlClient.request<{
+        groupByUrlname: {
+          upcomingEvents: { edges: { node: MeetupEventType }[] };
+        };
+      }>({
+        document: LAST_UPCOMING_EVENT_QUERY,
+        variables: {
+          urlname: import.meta.env.MEETUP_GROUP_ID,
+        },
+      });
+
+      const {
+        groupByUrlname: {
+          upcomingEvents: { edges },
+        },
+      } = data;
+
+      return edges.map((eventElement) => ({
+        ...eventElement.node,
+      }));
+    },
+  });
+}
+
+/**
+ * ### Usage
+ *
+ * ```ts
+ * import { fetchAllPastRomajsEvents } from '@api/meetup/queries.server';
+ *
+ * const allPastEvents = await fetchAllPastRomajsEvents();
+ * ```
+ */
+export async function fetchAllPastRomajsEvents(
+  config?: MeetupQueryConfig<MeetupEventType[]>
+): Promise<MeetupEventType[]> {
+  return queryCacheClient.fetchQuery({
+    ...config,
+    queryKey: ['fetchAllPastRomajsEvents'],
+    async queryFn() {
+      const itemsNum = await fetchRomajsMeetupsCount(
+        config as unknown as MeetupQueryConfig<number>
```

Comment:
It doesn't look like we are using these config parameters.

How do you feel about removing them?

If we want to write some tests it would keep our job easier.

Reply:
Those are all the configuration parameters to the [cache provider](https://tanstack.com/query/latest/docs/react/reference/useQuery).

Removed:
we ended up not using them and we can add them later, if we need to.